NebulaModel/Attributes/AttributeExtension.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Reflection;
+using System.Linq;
+using System;
+
+namespace NebulaModel.Attributes
+{
+    public static class AttributeExtension
+    {
+        public static T GetCustomAttribute<T>(this PropertyInfo property) where T : Attribute
+        {
+            return property.GetCustomAttributes(true)
+                .Select(att => att as T)
+                .Where(att => att != null)
+                .FirstOrDefault();
+        }
```

Comment:
Use MemberInfo for completeness and 
```suggestion
        public static T GetCustomAttribute<T>(this MemberInfo member) where T : Attribute
        {
            return member.GetCustomAttributes(true)
                .OfType<T>()
                .FirstOrDefault();
        }
```